src/database/FirestoreAccessor.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Iterator, List
+from enum import Enum
+
+import firebase_admin
+from firebase_admin import credentials
+from google.cloud import firestore
+
+from ..data.FilterCondition import FilterCondition
+from ..data.MetaDataItem import MetaDataItem
+from .iDatabase import iDatabase, AlreadyExistsException, NotExistingException
+from ..utils import get_project_root
+from .FirebaseAccessor import FirebaseAccessor
+
+FIREBASE_CRED_FILENAME = "autodash-9dccb-add3cdae62ea.json"
+
+
+# Raised if trying to instantiate Firebase Accessor without
+class MissingCredentials(Exception):
+    pass
+
+
+class QueryFilter(Enum):
+    NONE = 1
+    PROCESSED = 2
+    LABELLED = 3
+
+class FirestoreAccessor(iDatabase):
+
+    initialized = False
+
+    # Must be initialized only once
+    def initial_firebase(self):
+        dirname = get_project_root()
+        cred_file = os.path.join(dirname, "autodash-9dccb-add3cdae62ea.json")
+
+        if os.path.exists(cred_file):
+            # Have write access to Firebase
+            cred = credentials.Certificate(cred_file)
+            fb_app = firebase_admin.initialize_app(
+                cred, {
+                    'databaseURL': 'https://autodash-9dccb.firebaseio.com/',
+                    'databaseAuthVariableOverride': {
+                        'uid': 'pipeline-worker'
+                    }
+                })
+        else:
+            raise MissingCredentials()
+
+    def __init__(self, filter: QueryFilter = QueryFilter.NONE):
+        if not FirestoreAccessor.initialized and not FirebaseAccessor.initialized:
```

Comment:
Should update the check in FirebaseAccessor to do the same

Reply:
unfortunately I can't because then there's a circular dependency between the two accessors. However, there should only be one or the other initialized anyways (except for the migration which is why firestore has the check cause it appears after in the config), and once we transfer we can delete the firebase code